test/new_relic/telemetry_sdk/span_test.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require File.expand_path(File.join(File.dirname(__FILE__),'../..','test_helper'))
+
+require 'new_relic/telemetry_sdk/span'
+require 'new_relic/telemetry_sdk/util'
+
+module NewRelic
+  module TelemetrySdk
+    class SpanTest < Minitest::Test
+      def test_required_attributes
+        span = Span.new
+        assert span.id.is_a? String
+        assert span.trace_id.is_a? String
+        assert span.start_time_ms.is_a? Integer
+      end
+
+      def test_recommended_attributes
+        span = Span.new duration_ms: 123456,
+                        parent_id: "c617c2813a222a34",
+                        name: "Name",
+                        service_name: "My Service"
+
+        assert_equal 123456, span.duration_ms
+        assert span.duration_ms.is_a? Integer
+
+        assert_equal "Name", span.name
+        assert span.name.is_a? String
+
+        assert_equal "c617c2813a222a34", span.parent_id
+        assert span.parent_id.is_a? String
+
+        assert_equal "My Service", span.service_name
+        assert span.service_name.is_a? String
+      end
+
+      def test_custom_attributes
+        custom_attributes = {
+          :'user.email' => "me@newr.com",
+          :custom_key   => "custom_value"
+        }
+
+        span = Span.new custom_attributes: custom_attributes
+
+        assert_equal custom_attributes, span.custom_attributes
+      end
+
+      def test_finish_with_end_time_supplied
+        span = Util.stub :time_to_ms, 0 do
+          start_time_ms = Util.time_to_ms
+          Span.new start_time_ms: start_time_ms
+        end
+
+        Util.stub :time_to_ms, 1000 do
+          end_time_ms = Util.time_to_ms
+          span.finish end_time_ms: end_time_ms
+        end
+
+        assert_equal 1000, span.duration_ms
+      end
```

Comment:
For time related tests, have we considered using timecop to control the passage of time?

Reply:
Great question! I looked into using timecop but it does not support using monotonic time, which we plan to make an option via configuration. I would love to hear if you have thoughts on how we could test for both of these cases.

Reply:
Whoa, it looks like there is actually [a current PR](https://github.com/travisjeffery/timecop/pull/258) to add a monotonic time option to timecop! We should definitely keep an eye on that.

Other than that, we could change our test strategy conditionally, like in [this PR](https://github.com/DataDog/dogstatsd-ruby/pull/139).

Reply:
We should keep this in mind when we address configuration in #10.

Reply:
Added timecop to tests for now - we will need to do a conditional solution based on config when we add monotonic time.